kubernetes-client/src/main/java/io/fabric8/kubernetes/client/CustomResource.java

```diff
@@ -80,4 +127,85 @@ public ObjectMeta getMetadata() {
   public void setMetadata(ObjectMeta metadata) {
     this.metadata = metadata;
   }
+  
+  
+  public static String getPlural(Class<? extends CustomResource> clazz) {
+    final Plural fromAnnotation = clazz.getAnnotation(Plural.class);
+    return (fromAnnotation != null ? fromAnnotation.value().toLowerCase(Locale.ROOT) : Pluralize.toPlural(getSingular(clazz)));
+  }
+  
+  @JsonIgnore
+  public String getPlural() {
+    if(plural == null) {
+      final Plural fromAnnotation = getClass().getAnnotation(Plural.class);
+      this.plural = (fromAnnotation != null ? fromAnnotation.value().toLowerCase(Locale.ROOT) : Pluralize.toPlural(getSingular()));
+    }
+    return plural;
+  }
+  
+  public static String getSingular(Class<? extends CustomResource> clazz) {
+    final Singular fromAnnotation = clazz.getAnnotation(Singular.class);
+    return (fromAnnotation != null ? fromAnnotation.value().toLowerCase(Locale.ROOT) : HasMetadata.getKind(clazz)).toLowerCase(Locale.ROOT);
```

Comment:
```suggestion
    return (fromAnnotation != null ? fromAnnotation.value() : HasMetadata.getKind(clazz)).toLowerCase(Locale.ROOT);
```

Reply:
Shouldn't we make sure that the provided value is lower-cased?

Reply:
It's lowercased. I think that parentheses are confusing 😉